net/ser2net/files/etc/config/ser2net

```diff
@@ -0,0 +1,7 @@
+config 'proxy'
+        option 'tcpport' '8000'
+        option 'state' 'raw'
+        option 'timeout' '30'
+        option 'device' '/dev/ttyUSB0'
+        option 'options' '9600 1STOPBIT 8DATABITS'
```

Comment:
I think 2-stop bits is only ever used with 110 baud, and in any case the stop bits should be inferred from the baud rate if not given.  Likewise, 8-databits can be safely inferred unless something else is given.

Lastly, you don't seem to be including the parity (which, when absent, can be omitted and safely default to NONE which is the prevalent assumption in this post-Latin1/UTF8 world where we can't assume that everything is 7-bit ASCII).

Reply:
I can be mistaken but the default options is now set to 1STOPBIT? So your comment about 2-stop bits is not right?

Currently I put these patches in a pull-request and left the defaults as is. So if something can improved sure, no problem.

I personally don't know good defaults but I saw these: [9600 8DATABITS NONE 1STOPBIT] from here: https://www.domoticz.com/wiki/SharingSerialDevices is that ok?

Reply:
> I can be mistaken but the default options is now set to 1STOPBIT? So your comment about 2-stop bits is not right?

Not sure what you think isn't right about it.  I'm saying that 2 stopbits are never used except at 110 bps.

Most low-end serial devices are 38400.  Most high-end devices are 115200.

Reply:
Sorry for not understanding. But where do you see 2 stop bits? Currently the option is "1STOPBIT" that is 1 stop bit right?

Reply:
I'm saying in the `options`, only the baud rate should be required (if that).  The rest can be implied by the "normal" settings for any given seen (2-stopbits for 110 baud, 1-stopbit for anything else... no parity, 8-databits).

Reply:
Ok, will remove everything from options except for 9600.

Reply:
No, not what I'm asking for.

I'm saying if someone has:

`option options "9600"`

then you need to be able to synthesize the rest of the _implied_ values (parity, stopbits, databits).  Likewise, if someone has:

`option options "38400 even"`

then you need to synthesize stopbits and databits, etc.

I'd also support RTS/CTS and XON/XOFF, while you're at it.  Especially since https://github.com/lede-project/source/commit/7fe5963be023fd59a11652ecfb065d3405a91c48 was merged.

Reply:
Aah ok. Now I understand and partly agree.
Sure that would be a possibility, but is that really something we need to do? If such a thing is required shouldn't that be included in ser2net itself?

Because I fear that if we're going to autocomplete these options the filled in configuration != actual used configuration. I can imagine this could lead to misunderstanding when using ser2net.

Reply:
It's the other way around.

If `ser2net` changes its defaults, it shouldn't break OpenWRT installations that update the package.

By parsing the options and forcing all omitted values to *explicit* `ser2net` values, this can't possibly happen.

Reply:
But that will be an issue for any native ser2net installation.

But ok. I'm willing to update this. I've thought about 2 possible methods.

1. Dropdown with pre-filled in options and custom section so people can force their own.
2. Adding the additional option parameters in the init.d file during parsing.

The first one would be nice I suppose. Because then the configured options are identical to the resulting ser2net config file.

Reply:
If only 9600 is entered, just only put in 9600, let ser2net handle the rest.  Please don't go making asumptions on top of  what ser2net does.  @pprindeville yes, if ser2net changes its defaults, we _should_ get new defaults.  if you don't want to get defaults, be explicit.

Further, 2 stop bits is also used for modbus with no parity, to maintain 11bit words.  Don't make any assumptions about any of this please

Reply:
@TimelessNL Not sure what you mean by (1).  Are you talking about a GUI in LUA?

(2) sounds good to me.

@karlp I'm not making assumptions.  I'm going by the ECMA and ITU standards for RS-232.  If someone wants 2-stopbits they can explicitly call that out.  I'm just saying that for baud rates above 110, the same default is 1-stopbit.

Reply:
I don't like the current approach. Please change it to support the following config style:

    config proxy
        ...
        option baudrate <int>
        option databits <8|7|6|5>
        option parity <odd|even|none|space|mark>
        option stopbits <1|2>
        list options 'remctl'
        list options 'local'
        list options '-xonxoff'

Yes, this requires more work for generating the config since we need a mapping but I really think, this is the way we should go.

I agree with @karlp that we don't add assumptions here.

Reply:
@pprindeville Aah sorry about (1) I was already thinking ahead. I also got a pull-request for a LuCI-app so that's why I mentioned the dropdown :P. ofc this is invalid here.

@ both. Is it ok if I wait before doing anything until you 2 figured it out?

Reply:
The optional 4th argument to `config_get` is a default value.